### PR TITLE
create index on synclogstagingtable.user_id

### PR DESCRIPTION
**Pull request by snopoke**
##### SUMMARY
Index on user_id column speeds up load queries massively

### CODE REVIEW

File: corehq/warehouse/migrations/0032_auto_20190917_1542.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                choices=[(1, 'normal'), (2, 'archived'), (4, 'deprecated'), (8, 'duplicate'), (16, 'error'),
+                         (32, 'submission_error'), (64, 'deleted')]),
+        ),
+        migrations.AlterField(
+            model_name='formstagingtable',
+            name='state',
+            field=models.PositiveSmallIntegerField(
+                choices=[(1, 'normal'), (2, 'archived'), (4, 'deprecated'), (8, 'duplicate'), (16, 'error'),
+                         (32, 'submission_error'), (64, 'deleted')], default=1),
+        ),
```
Comment (snopoke) [inline corehq/warehouse/migrations/0032_auto_20190917_1542.py]: these other 2 changes are no-ops

File: corehq/warehouse/migrations/0032_auto_20190917_1542.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+from django.db import migrations, models
+
+TABLE_NAME = 'warehouse_synclogstagingtable'
+INDEX_NAME = 'warehouse_s_user_id_8bcfbf_idx'
```
Comment (mkangia) [inline corehq/warehouse/migrations/0032_auto_20190917_1542.py]: just curious about the `_s_` in the name?

**Review by mkangia — APPROVED** [review]

File: corehq/warehouse/migrations/0032_auto_20190917_1542.py (hunk 2 above)
Comment (snopoke) [reply corehq/warehouse/migrations/0032_auto_20190917_1542.py]: this is autogenerated by django. likely just a truncation `warehouse_synclogstagingtable`

Comment (mkangia) [reply corehq/warehouse/migrations/0032_auto_20190917_1542.py]: Thanks. I thought we write such migrations manually and hence i have always tried to come up with short unique name myself.

File: corehq/warehouse/migrations/0032_auto_20190917_1542.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+COLUMNS = ['user_id']
+
+
+CREATE_INDEX_SQL = "CREATE INDEX CONCURRENTLY IF NOT EXISTS {} ON {} ({})".format(
+    INDEX_NAME, TABLE_NAME, ','.join(COLUMNS)
+)
+DROP_INDEX_SQL = "DROP INDEX CONCURRENTLY {}".format(INDEX_NAME)
```
Comment (emord) [inline corehq/warehouse/migrations/0032_auto_20190917_1542.py]: Get excited for concurrent indexes natively supported https://docs.djangoproject.com/en/3.0/ref/contrib/postgres/operations/#index-concurrent-operations

**Pull request merged**